.gitignore

```diff
@@ -1,2 +1,6 @@
-build
+.build
```

Comment:
Because eventually we will need to add the bazel BUILD files, and some  filesystems are case insensitive.  Let me update the instructions.

Reply:
Good point! I didn't think about that.

Do you think it would make sense to use something other than `build` that doesn't start with a `.` so that it shows up in the `ls` output and easy to tab-complete? `.` will also tab-complete all the dotfiles, like `.gitignore`, `.dockerignore`, etc.

Bazel uses `bazel-*` directories; we can use `build-output`, `output`, `make-output`, `cmake-output`, or another name that would be easy to remember, easy to see, and easy to tab-complete. Thoughts?

Reply:
I see you added `.build` and `_build` to `.dockerignore` (because Windows can't have files that start with `.`, I believe?) but not `.gitignore` .

Would it make sense to adopt a name that could be identical across platforms, and would not start with a `.` or `_` instead?

Reply:
I think we should not overthink this: `build-output` sounds good enough.